Grafo.cpp

```diff
@@ -107,4 +107,170 @@ void Grafo::insertarArista(string origen, string destino, int precio, float tiem
+                int distancia = precioMatriz->obtenerValor(i,k) + precioMatriz->obtenerValor(k,j);
+
+                if(distancia < precioMatriz->obtenerValor(i,j))
+                    precioMatriz->modificarElemento(distancia,i,j); //Modifico la el valor
+
+                    aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j) //Modifico el aeropuerto
+
+            }
+        }
+    }
+    return recorridoMatriz;
+
+}
+
+
+Matriz<string>* Grafo::caminoMinimo(Matriz<float>* &tiempoMatriz)
+{
+    //Inicializo la matriz de Recorrido ...
+    Matriz<string>* recorridoMatriz = generarRecorrido();  ///LIBERAR DESPUES
+
+    //k = es el intermediario -> la letra de base
+    for(int k = 0 ; k < elementos ,k ++)
+    {
+        for(int i = 0 ;  i < elementos ; i ++)
+        {
+            for(int j = 0 ; j < elementos ; j++)
+            {
+                //Distancia acumulada
+                int distancia = tiempoMatriz->obtenerValor(i,k) + tiempoMatriz->obtenerValor(k,j);
+
+                if(distancia < tiempoMatriz->obtenerValor(i,j))
+                    tiempoMatriz->modificarElemento(distancia,i,j); //Modifico la el valor
+
+                    aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j) //Modifico el aeropuerto
+
+            }
+        }
+    }
+
+    return recorridoMatriz;
+}
+
+Lista<string>* Grafo::armarEscalas(int posicionOrigen , int posDestino , Matriz<string>* &recorridoMatriz)
+{
+    Lista<string>* escalas = new Lista<string> ; //Lista que posee las escalas
+
+    string escala = recorridoMatriz->obtenerValor(posicionOrigen,posicionDestino);
+
+    while(escala != NONE)
+    {
+        escalas->insertar(escala);
+        int posEscala = vertices->obtenerPosicion(escala);
+        escala =  recorridoMatriz->obtenerValor(posEscala , posicionDestino);
+    }
+
+    return escala;
+
+}
+
+void Grafo::minimoTiempo(string origen , string destino)
+{
+    Matriz<string>* recorridoMatriz = caminoMinimo(this->tiempoMatriz);
+
+
+    //Posiciones de los aeropuertos
+    unsigned posOrigen  = vertices->obtenerPosicion(origen);
+    unsigned posDestino = vertices->obtenerPosicion(destino);
+
+    int tiempoMinimo = tiempoMatriz->obtenerValor(posOrigen,posDestino); //Aca tengo el tiempo minimo entre el origen y destino
+
+    try {
+        if (tiempoMinimo == E_INFINITO)
+            throw (ExcepcionVuelo());
+
+
+        cout << "\nTiempo de vuelo:\t"<<tiempoMinimo<<"\n";
+        Lista<string>* escalas = armarEscalas(string origen , string destino, recorridoMatriz) ;
+
+        cout << "Escalas:";
+        for(int i = 0 ; i < escalas->obtenerTam() ; i++)
+        {
+            cout << "\t"<<escalas->obtenerDato(i);
+
+        }
+    }
+
+    catch(ExcepcionVuelo &e) {
+        cout << e.what() << endl;
+    }
+    //Elimino la memoria reservada (Matriz de recorrido y escalas)
+    delete escalas;
```

Comment:
Esto habría que hacerlo dentro del bloque `try`, justo después del ciclo `for` porque `escalas` solo existe en el bloque `try`. 
Además si movés la la declaración `Lista<string> *escalas;` afuera del bloque `try` (y adentrodel bloque haces `escalas = armarEscalas();`, va a haber un problema con este `delete` si se llegara a lanzar la excepción, porque no va a estar apuntando a ningún lugar de memoria ese puntero